src/providers/info.rs

```diff
@@ -76,6 +77,10 @@ impl InfoProvider for DefaultInfoProvider<'_> {
             anyhow::bail!("Token Address is 0x0")
         }
     }
+
+    fn raw_request(&mut self, url: &str) -> ApiResult<String> {
```

Comment:
I am not sure about this method .... I would prefer if the InfoProvider is very use case specific (so returns deserialized data)